### ISSUE TITLE
Sync with people is enabled when connected to the internet

### ISSUE THREAD

**Issue by gnarea** [relaycorp/relaynet-courier-android#105]
I just had a sneak peek at #98 and found this, so I'm not sure if it's a regression or it's already on master.

I started a sync with people and then stopped, and turned on WiFi. I'm not 100% certain of what happened next, but I think the option to sync with people became enabled (as expected) and then a few seconds later the option to sync with people became enabled. Either way, the option to sync with people shouldn't have been available:

![Screenshot (28 May 2020 20_25_24)](https://user-images.githubusercontent.com/369863/83184984-31854900-a122-11ea-81a8-aeb4d3b73da7.png)

```
2020-05-28 20:22:35.356 26801-26902/tech.relaycorp.courier I/UndispatchedCoroutine: Server started
2020-05-28 20:24:01.791 26801-27335/tech.relaycorp.courier W/nService$deliverCargo$1: deliverCargo error
    io.grpc.StatusRuntimeException: CANCELLED: cancelled before receiving half close
        at io.grpc.Status.asRuntimeException(Status.java:524)
        at io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onCancel(ServerCalls.java:275)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closedInternal(ServerCallImpl.java:353)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.closed(ServerCallImpl.java:341)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1Closed.runInContext(ServerImpl.java:868)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
2020-05-28 20:24:09.030 26801-26801/tech.relaycorp.courier I/CogRPCServer: Server stopped
2020-05-28 20:24:39.276 26801-26801/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 10
2020-05-28 20:24:39.337 26801-26801/tech.relaycorp.courier I/ifiHotspotStateReceiver: Wifi State 11
2020-05-28 20:24:51.864 26801-26801/tech.relaycorp.courier W/ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@37c8b7c
2020-05-28 20:25:01.009 26801-26801/tech.relaycorp.courier W/AboutLibraries: Have you missed to call withFields(R.string.class.getFields())? - autoDetect won't work - https://github.com/mikepenz/AboutLibraries/wiki/HOWTO:-Fragment
2020-05-28 20:25:01.034 26801-26801/tech.relaycorp.courier W/ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@a1aea88
```

**Comment by gnarea** [comment]
BTW, I ran into this issue again on my Pixel 2 this weekend but I was away from the computer so I couldn't check the logs.

I can't quite remember what I was doing before it happened, so take this with a pinch of salt: I think I was on the home screen of the app without access to the Internet, but then I opened another app and enabled WiFi. A few minutes later, I got back to the courier app and noticed the bug.
